### PR TITLE
add secret stats page for current milestones

### PR DESCRIPTION
**Pull request by staffanru**
A stats page added to easier track milestone goals
- at **/stats**

### CODE REVIEW

File: src/components/stats/h3-queries-stats.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        const hours: number[] = [];
+        const labels: string[] = [];
+        indata.forEach((e) => {
+            switch (e.is_valid) {
+                case 1:
+                    labels.push('Valid (hrs)');
+                    hours.push(e.hours);
+                    break;
+                case 2:
+                    labels.push('Invalid (hrs)');
```
Comment (judyfong) [inline src/components/stats/h3-queries-stats.tsx:46]: This label doesn't show up on the queries graph

File: src/components/stats/l2-speaker-stats.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    scales: {
+        yAxes: [
+            {
+                stacked: false,
+                ticks: {
+                    beginAtZero: true,
+                },
+                scaleLabel: {
+                    display: true,
+                    labelString: 'uttr',
```
Comment (judyfong) [inline src/components/stats/l2-speaker-stats.tsx:24]: I think utt is more common than uttr

File: src/components/stats/h3-queries-stats.tsx (hunk 1 above)
Comment (judyfong) [reply src/components/stats/h3-queries-stats.tsx:46]: Also the colors seems a bit wonky for the queries graph. "unverified" is green and "valid" is red

File: src/components/stats/repeat-audio-stats.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    const generateChart = () => {
+        const barChartData = generateBarChartData(data);
+        return <Bar data={barChartData} options={options} />;
+    };
+
+    const generateBarChartData = (indata: RepeatAudioStats[]) => {
+        const labels: string[] = [
+            'Total (uttr)',
+            'Valid (uttr)',
+            'Invalid (uttr)',
```
Comment (judyfong) [inline src/components/stats/repeat-audio-stats.tsx:44]: same here about uttr

File: src/components/stats/l2-speaker-stats.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    const { data, error } = useSWR('l2speaker', getL2SpeakersStats);
+
+    const generateChart = () => {
+        const barChartData = generateBarChartData(data);
+        return <Bar data={barChartData} options={options} />;
+    };
+    const generateBarChartData = (indata: L2SpeakerData) => {
+        const labels = ['Total (uttr)', 'Valid (uttr)', 'Invalid (uttr)'];
+        const barData = [indata.total, indata.valid, indata.invalid];
+        const goal = [50000, 0, 0];
```
Comment (judyfong) [inline src/components/stats/l2-speaker-stats.tsx:42]:
```suggestion
        const goal = [0, 50000, 0];
```

**Review by judyfong — CHANGES_REQUESTED** [review]
Looks great overall. I like that you use the h3 library. The graphs even include a goal so we know if we've reached it. However, there are a few small mistakes which need to be fixed. They're addressed by specific comments mostly.

I believe utt. is more common than uttr so please use that instead.

As a curiosity, why not add these graphs to stats.samromur.is instead of making a whole new page? that is somewhat secret although nothing is secret at all since this is a public repo

To be consistent with the file names in the frontend I believe we should us "herma" instead of "repeat-audio" or "repeat" for both the graphs and the filenames. But this is just a suggestion.

File: src/components/stats/h3-queries-stats.tsx (hunk 1 above)
Comment (staffanru) [reply src/components/stats/h3-queries-stats.tsx:46]: switched to a simpler implementation that actually works

File: src/components/stats/l2-speaker-stats.tsx (hunk 4 above)
Comment (staffanru) [reply src/components/stats/l2-speaker-stats.tsx:42]: I'll keep the goal for collected data as of now. Lets update this when we know how much we need to release. Then we might need to tweak these numbers to exclude data from other releases as well.

File: src/components/stats/l2-speaker-stats.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+const L2SpeakerStats: React.FunctionComponent = () => {
+    const { data, error } = useSWR('l2speaker', getL2SpeakersStats);
+
+    const generateChart = () => {
+        const barChartData = generateBarChartData(data);
+        return <Bar data={barChartData} options={options} />;
+    };
+    const generateBarChartData = (indata: L2SpeakerData) => {
+        const labels = ['Total (uttr)', 'Valid (uttr)', 'Invalid (uttr)'];
```
Comment (judyfong) [inline src/components/stats/l2-speaker-stats.tsx:40]:
```suggestion
        const labels = ['Total (utt)', 'Valid (utt)', 'Invalid (utt)'];
```

**Review by judyfong — CHANGES_REQUESTED** [review]
uttr -> utt

**Pull request merged**